### PR TITLE
feat: empty ui for create flows

### PR DESCRIPTION
**Pull request by tienifr**
### Details

This PR adds illustrations to empty list for create flows:
- Split Expense
- Submit Expense
- Pay Someone

### Fixed Issues

$ https://github.com/Expensify/App/issues/42227
PROPOSAL: https://github.com/Expensify/App/issues/42227#issuecomment-2115801897

### Tests

1. Create a new account
2. Press FAB >> Submit expense
3. Enter any amount and navigate to participant selector page
4. Verify loading skeleton appears first
5. Verify illustrations appear if the selection list is empty
6. Verify referral banner disappears
7. Search for any contacts
8. Verify illustrations disappear
9. Verify referral banner appears
5. Repeat with other flows as mentioned in Details

- [x] Verify that no errors appear in the JS console

### Offline tests

NA

### QA Steps

> [!NOTE]  
> Refer to **iOS: mWeb Safari** and **MacOS: Chrome / Safari** screenshots for the latest design.

1. Create a new account
2. Press FAB >> Submit expense
3. Enter any amount and navigate to participant selector page
4. Verify loading skeleton appears first
5. Verify illustrations appear if the selection list is empty
6. Search for any contacts
7. Verify illustrations disappear
10. Repeat with other flows as mentioned in Details

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

image

</details>

<details>
<summary>Android: mWeb Chrome</summary>

image

</details>

<details>
<summary>iOS: Native</summary>

image

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/fb5b04f5-e764-41c6-b784-4cac0f97c8b8

Simulator Screenshot - iPhone 15 Pro Max - 2024-07-24 at 16 17 36

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

Screenshot 2024-07-24 at 16 15 40

</details>

<details>
<summary>MacOS: Desktop</summary>

Screenshot 2024-07-24 at 16 18 49

</details>

### CODE REVIEW

**Comment by shawnborton** [comment]
From a copy perspective, it looks like we have an orphan in "number" - cc @jamesdeanexpensify

**Comment by shawnborton** [comment]
Looks like the illustration is cut off on iOS:
CleanShot 2024-05-21 at 07 33 12@2x

**Comment by jamesdeanexpensify** [comment]
@shawnborton tagged ya [here](https://expensify.slack.com/archives/C03TBKRSB/p1716309586218859?thread_ts=1714766113.560229&cid=C03TBKRSB) to get alignment.

**Review by rayane-d — CHANGES_REQUESTED** [review]

**Comment by jamesdeanexpensify** [comment]
@danielrvidal we're going with the updated copy below as decided [here](https://expensify.slack.com/archives/C03TBKRSB/p1716488406814139?thread_ts=1714766113.560229&cid=C03TBKRSB), right?

> **Submit an expense**
> Submit to someone and [get $250](https://help.expensify.com/articles/expensify-classic/expensify-partner-program/Referral-Program) when they become a customer.

> **Split an expense**
> Split with a friend and [get $250](https://help.expensify.com/articles/expensify-classic/expensify-partner-program/Referral-Program) when they become a customer.

> **Pay someone**
> Pay anyone and [get $250](https://help.expensify.com/articles/expensify-classic/expensify-partner-program/Referral-Program) when they become a customer.

**Comment by dubielzyk-expensify** [comment]
> What do you think it should be like?

My take is that the illustration and text should ignore the keyboard and be centered vertically on the. That means that when the keyboard moves in it'll obscure parts of the text and illustration.

**Comment by shawnborton** [comment]
Yup, I agree with @dubielzyk-expensify in that the keyboard would sit on top of the content in this case. 

Also, I think we decided to only show the referral banner once you start seeing results in the contact list after searching? This way we aren't duplicating the same message from the empty state.

**Comment by dannymcclain** [comment]
>My take is that the illustration and text should ignore the keyboard and be centered vertically
>That means that when the keyboard moves in it'll obscure parts of the text and illustration.

Yup, I agree too.

**Comment by jamesdeanexpensify** [comment]
Just double checking - where should I confirm that we're updating to [this copy](https://github.com/Expensify/App/pull/42413#issuecomment-2129859022)? Thank you!

**Comment by danielrvidal** [comment]
@tienifr can you please make sure to update the copy and screenshots to reflect this [comment from](https://github.com/Expensify/App/pull/42413#issuecomment-2129859022) @jamesdeanexpensify.  

That matches the [copy here](https://github.com/Expensify/App/issues/42227#issuecomment-2129970562), where you can also find the Spanish copy as well.

**Comment by allroundexperts** [comment]
Lint failing @tienifr

**Comment by shawnborton** [comment]
Actually rather than forcing a line break, what if we just added like 20px of padding extra horizontally?

**Comment by shawnborton** [comment]
Or actually, I would also be fine just doing nothing so that we aren't making any special one-off exceptions here. Maybe that's the best path forward.

**Comment by dubielzyk-expensify** [comment]
Also looks like that on mWeb iOS the promo banner isn't at the bottom:
CleanShot 2024-06-03 at 08 39 45@2x

**Comment by allroundexperts** [comment]
Conflicts @tienifr

**Comment by allroundexperts** [comment]
@tienifr What's the latest here?

**Comment by shawnborton** [comment]
> The keyboard would sit on top of the content in this case
>  image


Interesting... because don't we auto-focus that field every time too? I guess that means on mobile, this would always be covered up no matter what, right?

**Comment by shawnborton** [comment]
Hah good timing indeed!

So maybe we can see a version where the content shrinks when the keyboard is up (as in, the keyboard does not sit on top of the content), but we can give that view a min-height so it doesn't look too bad on the shorter mobile devices which would allow the keyboard to partially cover it if needed?

**Comment by shawnborton** [comment]
Not opposed to that, but I say let's first start by just seeing what it looks like with my suggestion [here](https://github.com/Expensify/App/pull/42413#issuecomment-2222334243) and then we can adjust to the smaller device sizes from there.

**Comment by shawnborton** [comment]
Okay cool, I think it's worth trying to make the illustration slightly smaller too. Could you try it at 146x148 and see if that looks nice?

**Comment by shawnborton** [comment]
Can you show a similar spread of screenshots for a more modern iPhone too for comparison? I don't think we should overly optimize for SE, though what you have for 0x0 is interesting - we could consider just not showing an illustration when the screen height is quite small. Thoughts @Expensify/design?

**Comment by dannymcclain** [comment]
I agree that we probably don't need to overly optimize for super tiny viewports. Not against ditching the illustration for extra teeny tiny screens, but personally I also don't mind if the user has to scroll to see the message (that might just be me though!)

I do kinda want a _little_ bit more space between the input and the illustration though. It feels pretty tight to me.

**Comment by shawnborton** [comment]
Cool, I think that's good feedback. @tienifr can you adjust for us so we can see another round of mocks?

**Comment by shawnborton** [comment]
Cool, thanks for sharing! I think at this point I would be down to just use something like `120x125` everywhere, and not worry about any kind of conditional sizes for smaller screens (because as Danny mentioned, you can always scroll to see more). Let's see what the rest of the @Expensify/design team thinks though!

**Comment by dannymcclain** [comment]
>I think at this point I would be down to just use something like 120x125 everywhere, and not worry about any kind of conditional sizes for smaller screens

Definitely down for this approach!

**Comment by mountiny** [comment]
@tienifr sounds like we got an agreement on how to move this forwards, can you please implement it? thanks!

**Comment by dubielzyk-expensify** [comment]
Awesome breakdown. I agree that `120x125` will work well for all use cases 👍

**Comment by mountiny** [comment]
sooooo, Are we good for final review @allroundexperts?

**Comment by allroundexperts** [comment]
Yes. On it again now!

**Comment by allroundexperts** [comment]
@tienifr It seems like you've not pushed your changes yet. Since we have an agreement to use 120x125 as the size, I think this would be fairly simple to implement.

**Comment by mountiny** [comment]
@allroundexperts what is your ETA to complete the checklist here?

**Comment by allroundexperts** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>


https://github.com/user-attachments/assets/2979e55e-2b08-4d99-8b81-2cead23c8b39

</details>

<details>
<summary>Android: mWeb Chrome</summary>


https://github.com/user-attachments/assets/400dd434-7a28-4376-856a-3715cbe1b919

</details>

<details>
<summary>iOS: Native</summary>


https://github.com/user-attachments/assets/e3d69b0b-e07c-4cf0-b778-a94c4aa0777a

</details>

<details>
<summary>iOS: mWeb Safari</summary>


https://github.com/user-attachments/assets/6e3a9bcf-0bc1-4375-9092-64c753f78173

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>


https://github.com/user-attachments/assets/346239fc-e1e1-458a-851b-b3f39a35b045

</details>

<details>
<summary>MacOS: Desktop</summary>


https://github.com/user-attachments/assets/87d7be6d-a20a-4733-a146-703f27a0d7c6

</details>

**Review by allroundexperts — APPROVED** [review]
Looks good to me!

@Expensify/design Can you give a design approval?

@tienifr Are the translations approved?

**Comment by dubielzyk-expensify** [comment]
There's a slight jump when there's no results found. I think we should make sure this jump doesn't happen. It's a bit jarring:
CleanShot 2024-07-29 at 13 47 16

Also, I think we should add extra padding on the bottom of like 20-32px to make sure that the illustration and text looks more centered vertically. (It might technically be centered, but if often looks too low in these instances, so adding a bit of padding will help:
CleanShot 2024-07-29 at 13 50 05@2x

But keen to hear if @Expensify/design agrees here

**Comment by shawnborton** [comment]
Those are great comments, I agree with both of them.

**Comment by shawnborton** [comment]
For the "not found" jump - I'm guessing that's not what Jon had in mind. I think the idea is to always reserve space below the input for the not found message, so that this way if the message is there, it doesn't cause the illustration below to move. Let me know if that makes sense. 

Unrelated, why is there so much green at the top/bottom of this view?
CleanShot 2024-08-01 at 08 10 07@2x

**Comment by mountiny** [comment]
@dubielzyk-expensify Would you be able to check this out please? 

@tienifr seems like lint is failing

**Comment by dubielzyk-expensify** [comment]
Yeah, the behavior I think we should aim for is exactly what Shawn mentioned above:

>  I think the idea is to always reserve space below the input for the not found message, so that this way if the message is there, it doesn't cause the illustration below to move.

**Comment by mountiny** [comment]
@tienifr can you please do the final updates here? thanks!

**Comment by mountiny** [comment]
@tienifr can you please prioritize this one to get completed? Thanks!

**Comment by shawnborton** [comment]
Hmm I think we're still not on the same page here. The idea is to keep the illustration when there are no results found, but we should make it so that the "no results found" message does not cause the illustration to jump. Let me know if that makes sense.

**Comment by shawnborton** [comment]
> If we decide to show the illustration in case 1 and 2, it can cause the illustration to jump.

Yeah, so the suggestion is to leave a fixed space below the input so that when the "No results found" message appears, the illustration doesn't jump. Does that make sense?

**Comment by mountiny** [comment]
@shawnborton do you think you could whip a quick figma gif or something to cross the imagination gap here? Thanks!

**Comment by dubielzyk-expensify** [comment]
> There're 3 cases:
>
> 1. At the beginning when users don't type anything yet, there's no result and we don't show no results found messages.
> 2. Users search someone -> there's no result and we show no results found messages.
> 3. Users search someone -> there're some results and we don't show no results found messages.
>
> Currently, we show illustration in case 1 only
>
> If we decide to show the illustration in case 1 and 2, it can cause the illustration to jump.

We want the illustration to stay at the same place regardless of whether the `No results found` message is showing. And we want it for both case 1 and 2 listed above. 

Previously it caused jumping because when the string `No results found` was added, it caused the illustration to move down slightly. 

So here in the illustration you can see the two states, `Empty` and `No Results Found`. In both instances the illustration stays in the same place and the only difference is just that the `No results found` string appears below the input.

CleanShot 2024-08-13 at 10 04 48@2x



Hopefully that helps and thank you for your patience on this one @tienifr !

**Comment by shawnborton** [comment]
Yes! Thank you so much Jon.

**Comment by dubielzyk-expensify** [comment]
Looooovely! Can we move the illustration up about 20-32px? It sits a bit low. Otherwise this is awesome work 😄  Thank you!

**Comment by shawnborton** [comment]
Looking good! Thanks for bearing with our feedback. 

One small idea as I see this - should we slightly increase the horizontal padding of the empty state so that the text wraps in a more friendly way? Basically so that we can get "become..." to start on the second line, which might balance out how that sentence looks split across the two lines?
CleanShot 2024-08-13 at 05 58 07@2x

**Comment by dannymcclain** [comment]
>One small idea as I see this - should we slightly increase the horizontal padding of the empty state so that the text wraps in a more friendly way?

I like that idea. I also wonder if instead of changing the padding, we just add a line break so it wraps exactly how we want it to?

**Comment by shawnborton** [comment]
True... we could definitely try that out and see what happens!

**Comment by shawnborton** [comment]
Looks good!

**Review by mountiny — APPROVED** [review]
Thank you for the patience @tienifr lets be on a look out for any regressions

**Comment by mountiny** [comment]
given Shawn's approval I will move this ahead to avoid any merge conflicts

**Comment by shawnborton** [comment]
Hmm I am getting some jumpiness when the "not found" message appears under the search input - I thought we fixed that?

https://github.com/user-attachments/assets/67254a2c-ba06-484c-9770-0817c4ba3682

**Comment by mountiny** [comment]
@tienifr Are you able to look into this one. Can you reproduce? It should have been fixed I believe

File: src/components/SelectionList/BaseSelectionList.tsx [hunk 1]
```diff
@@ -672,14 +682,14 @@ function BaseSelectionList<TItem extends ListItem>(
...
                         <View style={headerMessageStyle ?? [styles.ph5, styles.pb5]}>
                             <Text style={[styles.textLabel, styles.colorMuted]}>{headerMessage}</Text>
                         </View>
                     )}
                     {!!headerContent && headerContent}
-                    {flattenedSections.allOptions.length === 0 && showLoadingPlaceholder ? (
-                        <OptionsListSkeletonView shouldStyleAsTable={shouldUseUserSkeletonView} />
+                    {flattenedSections.allOptions.length === 0 ? (
+                        renderListEmptyContent()
                     ) : (
```
Comment (hungvu193) [inline src/components/SelectionList/BaseSelectionList.tsx:693]: This line introduces https://github.com/Expensify/App/issues/50577

More details in here: https://github.com/Expensify/App/issues/50577#issuecomment-2405431272

**Pull request merged**